csharp/src/Ice/OutputStream.cs

```diff
@@ -771,12 +779,17 @@ public void WriteTaggedSeq<T>(int tag, IEnumerable<T>? v, int elementSize, Outpu
         /// <param name="v">The sequence to write.</param>
         /// <param name="elementSize">The fixed size of each element of the sequence, in bytes.</param>
         /// <param name="writer">The delegate that writes each element to the stream.</param>
-        public void WriteTaggedSeq<T>(int tag, IEnumerable<T>? v, int elementSize,
-            OutputStreamStructWriter<T> writer) where T : struct, IStreamableStruct
+        public void WriteTaggedSequence<T>(int tag,
+                                          IEnumerable<T>? v,
+                                          int elementSize,
+                                          OutputStreamStructWriter<T> writer)
+            where T : struct, IStreamableStruct
         {
             Debug.Assert(elementSize > 0);
-            if (v is IEnumerable<T> value && WriteOptional(tag, OptionalFormat.VSize))
+            if (v is IEnumerable<T> value)
             {
+                WriteTaggedParamHeader(tag, OptionalFormat.VSize);
+
```

Comment:
seems this could call WriteSequence too instead of using foreach

Reply:
I added a comment: the WriteSequence is "inline" to avoid recomputing the count, which can be occasionally expensive.